### ISSUE TITLE
CI never exercises the shaded JAR, so packaging regressions pass green

### ISSUE THREAD

**Issue by mosser** [jpipe-mcscert/jpipe-compiler#153]
Nothing in the build runs the fat JAR. Tests execute against each module's own classpath, so a dependency-scope or shading mistake is invisible to `mvn verify` even though the shaded JAR is the artifact users actually run.

## How this surfaced

While reviewing poms during #152, `org.json` was briefly marked `<scope>test</scope>` in `jpipe-cli`. A direct declaration overrides the transitive scope, so the shade plugin dropped the library from the JAR. `mvn verify` stayed **fully green** — 717 tests, 0 checkstyle violations — while the shipped CLI was broken:

```
$ java -jar jpipe-cli.jar diagnostic -i examples/000_minimal.jd -f json
Exception in thread "main" java.lang.NoClassDefFoundError: org/json/JSONObject

$ java -jar jpipe-cli.jar process -i examples/000_minimal.jd -m minimal -f json
Exception in thread "main" java.lang.NoClassDefFoundError: org/json/JSONArray
```

That took out the pre-existing model export as well, not just the new JSON report. The change was reverted, so `dev` is fine — but only because it was caught by hand.

## Why tests cannot catch it

The gap is structural rather than a missing test case:

- surefire runs against `target/classes` plus the module's resolved dependencies;
- the shaded JAR is assembled afterwards by `maven-shade-plugin` from the **runtime** classpath of `jpipe-cli`;
- nothing then loads that JAR.

Anything that only manifests after shading — a missing transitive dependency, a relocation, a merged/overwritten `META-INF` service file, a missing packaged resource such as `/schema/diagnostic-report-v1.schema.json` — is invisible.

## Suggested fix

A smoke-test step in `build.yml` after `mvn package`, running the built JAR once per output path and asserting exit code and shape rather than exact bytes. Roughly:

```bash
JAR=$(ls jpipe-cli/target/jpipe-cli-*.jar | grep -v original)
java -jar "$JAR" --headless diagnostic -i examples/000_minimal.jd            > /dev/null
java -jar "$JAR" --headless diagnostic -i examples/000_minimal.jd -f json | jq -e .schemaVersion > /dev/null
java -jar "$JAR" --headless process    -i examples/000_minimal.jd -m minimal -f json | jq -e .name > /dev/null
java -jar "$JAR" --headless process    -i examples/000_minimal.jd -m minimal -f dot  > /dev/null
java -jar "$JAR" doctor                                                       > /dev/null
```

Worth also asserting the negative cases the JAR is supposed to get right, since both are packaging-sensitive:

- `diagnostic -f json` on a clean file exits 0, on a file with validation errors exits 1;
- `diagnostic -f json` to stdout emits parseable JSON **without** `--headless` (the logo suppression added in #152).

Two smaller options if a CI step is unwanted: bind the checks to the `verify` phase with `exec-maven-plugin` so they also run locally, or add a `maven-invoker-plugin` integration test. The CI step is the cheapest and covers the case that actually bit us.

## Notes

- Graphviz is already installed in `build.yml`, so image formats could be included, though they are slower and less packaging-sensitive.
- Keep it to a handful of invocations; the point is to prove the JAR loads and its main paths run, not to duplicate the unit and Cucumber suites.
